LinkedList/Exercise/SLList.h

```diff
@@ -0,0 +1,328 @@
+
+	FNode* GetNode(const int Index)
+	{
+		FNode* Current = Head;
+		for (int i = 0; i < Index; i++)
+		{
+			Current = Current->Next;
+		}
+		return Current;
+	}
+
+public:
+
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++ ()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator* ()
+		{
+			return Current->Element;
+		}
+
+		bool operator!= (const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+
+	};
+
+	FIterator begin()
+	{
+		if (Head)
+		{
+			return FIterator(Head);
+		}
+		return FIterator(nullptr);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+
+
+	//Constructor
+	//-------------------------------------------------------------------------------------
+
+	TSLList() : Head(nullptr), Tail(nullptr), ListSize(0) {};
+
+	//-------------------------------------------------------------------------------------
+
+	~TSLList()
+	{
+		Clear();
+	};
+
+	//-------------------------------------------------------------------------------------
+
+	TSLList(TSLList& InList)
+	{
+		FNode* Current = InList.Head;
+		for (int i = 0; i < InList.ListSize; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& operator[] (const int Index)
+	{
+		return GetNode(Index)->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void operator= (TSLList& InList)
+	{
+		Clear();
+		FNode* Current = InList.Head;
+		for (int i = 0; i < InList.ListSize; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	const Type& operator= (const int Index) const
+	{
+		return GetNode(Index)->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& GetHead()
+	{
+		return Head->Element;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& GetTail()
+	{
+		return Tail->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	const Type& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	bool IsEmpty() const
+	{
+		return ListSize == 0;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	int GetSize() const
+	{
+		return ListSize;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void AddHead(const Type& InElement)
+	{
+		FNode* NewHead = new FNode();
+		NewHead->Element = InElement;
+		NewHead->Next = Head;
```

Comment:
Since you're gonna' be instantiating lots of FNodes you might want to add a constructor for these that takes in Element and Head to save you lines of code

Reply:
great advice 👌 thx